### PR TITLE
Allow advanced guessit configuration

### PR DESCRIPTION
**Pull request by ratoaq2**
Work in progress...

My goal here is to start a discussion, since there are a few things that doesn't fell right to me:
- The advanced rules configuration is on the same internal options file. That causes issues if you specify `--no-embedded-config` since those advanced configurations are needed to build rebulk rules. The other issue is every guessit call, the options object is always transversed to fix the encoding and all the advanced configuration was only used in the very first time the rebulk object was created.
- Some tests are still failing
- Maybe not all of those configuration should be externalized
- Common words still need to be handled

### CODE REVIEW

File: guessit/api.py [hunk 1]
```diff
@@ -42,6 +42,17 @@ def __init__(self, string, options):
         self.options = options
 
 
+def load_rules(options):
+    """
+    Load rebulk rules according to rules in options dictionary.
```
Comment (ratoaq2) [inline guessit/api.py]: Fix wording here

**Comment by ratoaq2** [comment]
Now it seems ready.

- Guessit rules are configured only in the first call to `guessit.guessit(...)`
- Since the advanced configuration is required to compile and configure guessit internal rules, the default advanced configuration is properly loaded even if `--no-embedded-config` is defined. And it only happens once, but it can be forced using the new api method: `guessit.configure(options)`

**Comment by ratoaq2** [comment]
Now season in other languages are configurable. It will close #277 as well.

**Comment by ratoaq2** [comment]
@Toilal if you agree with this PR, we can merge it and create a release candidate for guessit 3.0

@Diaoul, guessit 3.0 is a major release with some breaking changes ( https://github.com/guessit-io/guessit/blob/develop/HISTORY.rst ). Do you think guessit 3.0 rc1 release will break subliminal? I can see that in subliminal you have `guessit>=2.0.1`. Should we coordinate this?

**Comment by medariox** [comment]
Any news about this? We are impatiently waiting for it 🎉

**Comment by Toilal** [comment]
I'll give a try to this tonight, and then merge for guessit 3 release. Anything to add @ratoaq2 ?

**Comment by Toilal** [comment]
It seems really good to me, great job !

**Comment by dimotsai** [comment]
@ratoaq2 👍 

So is `advanced_config` still not configurable through `--config` option?
It seems like the default `advanced_config` overrides custom `advanced_config` anyway.

**Comment by ratoaq2** [comment]
@Toilal, it seems http://guessit.readthedocs.io/en/latest/ is not updated to the latest

**Pull request merged**